lib/ymodel/lib/ymodel/loadable.rb

```diff
@@ -21,11 +21,40 @@ def schema
       YModel::Schema.new({})
     end
 
+    def load_records!
+      all = records.map { |record| new(record) }
+      unless all.map(&:index) == all.map(&:index).uniq
+        raise YModel::DuplicateIndexError, 'Some records share the same index.'
+      end
+
+      all
+    end
+
+    def index_on(key)
+      @index = key
+    end
+
+    def index
+      @index || :id
+    end
+
     protected
 
+    # I dislike the format of this method a lot. Rubocop made me do it..
```

Comment:
I really dislike the way this method is factored. I would love to get suggestions about how to improve it without getting scolded by the linter .